metis-indexing/src/main/java/eu/europeana/indexing/tiers/RecordTierCalculationViewGenerator.java

```diff
@@ -0,0 +1,91 @@
+  private final String stringRdf;
+
+  public RecordTierCalculationViewGenerator(String europeanaId, String providerId, String stringRdf) {
+    this.europeanaId = europeanaId;
+    this.providerId = providerId;
+    this.stringRdf = stringRdf;
+  }
+
+  public RecordTierCalculationView generate() {
+    //Create an object that has predefined "fake" values
+    RecordTierCalculationView recordTierCalculationView = FakeTierCalculationProvider.getFakeObject();
+    final RecordTierCalculationSummary recordTierCalculationSummary = recordTierCalculationView.getRecordTierCalculationSummary();
+    final TierValues tierValues = calculateTierValues(stringRdf);
+
+    //Update only the summary which is what we need for now
+    // TODO: 04/01/2022 Update the remaining values
+    recordTierCalculationSummary.setEuropeanaRecordId(europeanaId); // Here use XmlRecordProcessorServiceImpl
+    recordTierCalculationSummary.setProviderRecordId(providerId);
+    recordTierCalculationSummary.setContentTier(tierValues.getContentTier());
+    recordTierCalculationSummary.setMetadataTier(tierValues.getMetadataTier());
+    recordTierCalculationSummary.setPortalLink("https://example.com"); // We need the root link from the configuration
+    recordTierCalculationSummary.setHarvestedRecordLink("https://example.com"); // We need a proxy controller for this
+
+    return recordTierCalculationView;
+  }
+
+  private TierValues calculateTierValues(String xml) {
+    final RDF rdf;
+    try {
+      // Perform the tier classification
+      rdf = RdfConversionUtils.convertStringToRdf(xml);
+      final RdfWrapper rdfWrapper = new RdfWrapper(rdf);
+      RdfTierUtils.setTier(rdf, ClassifierFactory.getMediaClassifier().classify(rdfWrapper));
+      RdfTierUtils.setTier(rdf, ClassifierFactory.getMetadataClassifier().classify(rdfWrapper));
+
+      final RdfToFullBeanConverter fullBeanConverter = new RdfToFullBeanConverter();
+      final FullBeanImpl fullBean = fullBeanConverter.convertRdfToFullBean(rdfWrapper);
+      final List<SolrTier> solrTiers = fullBean.getQualityAnnotations().stream().map(RdfTierUtils::getTier)
+          .map(RdfTierUtils::getSolrTier).collect(
+              Collectors.toList());
+      final String contentTier = solrTiers.stream().filter(solrTier -> solrTier.getTierLabel() == EdmLabel.CONTENT_TIER)
+          .findFirst().map(SolrTier::getTierValue).orElse(null);
+      final String metadataTier = solrTiers.stream().filter(solrTier -> solrTier.getTierLabel() == EdmLabel.METADATA_TIER)
+          .findFirst().map(SolrTier::getTierValue).orElse(null);
+      return new TierValues(contentTier, metadataTier);
+    } catch (SerializationException | IndexingException e) {
+      throw new TierCalculationException("Error during calculation of tiers", e);
+    }
+  }
+
+  static class TierValues {
+
+    private final String contentTier;
+    private final String metadataTier;
+
+    public TierValues(String contentTier, String metadataTier) {
```

Comment:
same here. ```final``` parameters.

Reply:
Not adding it here.
Check https://github.com/europeana/metis-framework/pull/493#discussion_r781886073